### ISSUE TITLE
The scores of different model

### ISSUE THREAD

**Issue by daralevian** [HansBambel/SmaAt-UNet#6]
I'sorry to border again! I've successfully get the pictures through the ipynb file which similar to the article,
but the result of the test_precip_lightning.py is like this:
![image](https://user-images.githubusercontent.com/107391441/232835531-3e66b59a-17cb-4e55-a633-e027cb793c9e.png)
There're only one model's scores
I'd like to ask how to get all the scores in the article
![image](https://user-images.githubusercontent.com/107391441/232835947-f95e3d43-173f-486d-a9df-ad319e55e043.png)
I don't know if the result is set to be like the first one or is there any problem to the code.
Thank you so much!

**Comment by HansBambel** [comment]
It seems like you did not place the models that you want to compare in the comparison folder that was mentioned in the other issues #3 #5 
That's why you only receive the loss for the persistence model (which gets calculated by default).
As for the other point: I do not know exactly why you receive different results for that model.

**Comment by daralevian** [comment]
![image](https://user-images.githubusercontent.com/107391441/232843419-e695d11f-ac43-4ffa-a9dc-ed2075f9a3e4.png)
I put them like this
and the results of ipynb is like this 
![image](https://user-images.githubusercontent.com/107391441/232843878-585fd84d-99a0-4b33-b5c9-a00cce951272.png)

**Comment by daralevian** [comment]
![image](https://user-images.githubusercontent.com/107391441/232844762-3f59903a-28ba-44ba-a7c0-0f5b3621bf75.png)
I put them like this
and the picture is like this:
![image](https://user-images.githubusercontent.com/107391441/232845019-a719d8c2-757a-49e1-9362-f4600f8e800b.png)

**Comment by HansBambel** [comment]
Interesting, so the plotting works, but the loss-calculation does not.

You can add `print(os.listdir(model_folder))` in line 135 of `test_precip_lightning.py` to check what models are visible at runtime for debugging. From where do you execute the script? It could be that the absolute path of the model folder is somewhere different you expect.

I will open an issue about that and use the Path module to fix this when I have time.

**Comment by daralevian** [comment]
I tried ,and it response like this:
![image](https://user-images.githubusercontent.com/107391441/232963236-8cde49cf-3113-4769-ab5c-2bca90c89ef2.png)

**Comment by HansBambel** [comment]
Ah, I see. Everything works correctly. The losses are also calculated. They are plotted. You probably only want them to print, right? 
You now have a pkl file with the model losses. In there is a dict with the model losses that you can use.

**Comment by daralevian** [comment]
OK,I got it. Thank you so much!!

**Comment by codeGotham** [comment]
> ![图像](https://user-images.githubusercontent.com/107391441/232844762-3f59903a-28ba-44ba-a7c0-0f5b3621bf75.png)我把它们放成这样，图片是这样的： ![图像](https://user-images.githubusercontent.com/107391441/232845019-a719d8c2-757a-49e1-9362-f4600f8e800b.png)
I can‘t find your e-mail 
sorry to border， can  I ask you some question in private by  e-mail？I can‘t find your e-mail

**Comment by HansBambel** [comment]
@codeGotham Is there a question?

**Comment by daralevian** [comment]
@HansBambel I try to resd the pkl file ,and it results like this ,is it right?
![image](https://user-images.githubusercontent.com/107391441/233834130-4aa82765-b11c-4734-8d51-2e3c769f3dad.png)

**Comment by daralevian** [comment]
> > ![图像](https://user-images.githubusercontent.com/107391441/232844762-3f59903a-28ba-44ba-a7c0-0f5b3621bf75.png)我把它们放成这样，图片是这样的： ![图像](https://user-images.githubusercontent.com/107391441/232845019-a719d8c2-757a-49e1-9362-f4600f8e800b.png)
> > I can‘t find your e-mail
> > sorry to border， can  I ask you some question in private by  e-mail？I can‘t find your e-mail

ok ,my e-mail is 727416400@qq.com

**Comment by HansBambel** [comment]
Looks about right :)

But the values seem to be quite different than in the paper. I do not know exactly why that is the case. Could be that you have a different Pytorch version or something like that or chose a different checkpoint file.

**Comment by daralevian** [comment]
> Looks about right :)
> 
> But the values seem to be quite different than in the paper. I do not know exactly why that is the case. Could be that you have a different Pytorch version or something like that or chose a different checkpoint file.

ok,i'll try to change it ,but there's only one score about the model ,how do I print other scores like accuracy, sci?

**Comment by daralevian** [comment]
> Looks about right :)
> 
> But the values seem to be quite different than in the paper. I do not know exactly why that is the case. Could be that you have a different Pytorch version or something like that or chose a different checkpoint file.

yeah , but the pictures are really similar，so I‘m so tortured by the scores.

**Comment by HansBambel** [comment]
You are right, I forgot to add the script for calculating those metrics. I just added it: [calc_metrics_test_set.py](calc_metrics_test_set.py). 

I think I will refactor some code after setting up the environment when I find the time.

**Comment by daralevian** [comment]
> You are right, I forgot to add the script for calculating those metrics. I just added it: [calc_metrics_test_set.py](calc_metrics_test_set.py).
> 
> I think I will refactor some code after setting up the environment when I find the time. Then all losses and metrics will be calculating together.

OK, I got it, so can you remind me if the work has done, please? Thank you so much!!!!!!!!

**Comment by daralevian** [comment]
> Will do, but until then you can use the script already as well.
> 
> The refactor will include upgrading the packages such as pytorch and lightning so I am not sure if the experiments will be reproducible. I made a snapshot of the code from the paper when it was published so that people can try to reproduce the experiments https://github.com/HansBambel/SmaAt-UNet/tree/snapshot-paper

ok,I got it. thanks

**Comment by HansBambel** [comment]
I started an port: https://github.com/HansBambel/SmaAt-UNet/pull/8

This is still a WIP, so not everything works and it could be that the experiments are not 100% reproducible. So, for reproducing the beforementioned branch is necessary.

**Comment by daralevian** [comment]
sorry to boder again, could I ask about the code about calculate MSE and NMSE ? I didn't find them in the code I use @HansBambel

**Comment by HansBambel** [comment]
The loss is already NMSE:
```
loss_func(y_pred.squeeze() * factor, y_true * factor, reduction="sum") / y_true.size(0)
```
To get the MSE I think you need to do this:
```
loss_func(y_pred.squeeze() * factor, y_true * factor, reduction="mean")
```
